addons/io_hubs_addon/components/operators.py

```diff
@@ -649,23 +650,23 @@ def execute(self, context):
         old_img = self.hubs_component[self.target_property]
 
         # Load/Reload the first image and assign it to the target property, then load the rest of the images if they're not already loaded. This mimics Blender's default open files behavior.
-        primary_filepath = os.path.join(dirname, self.files[0].name)
```

Comment:
This line still needs to be here to match Blender's default behavior when opening multiple files, just replacing dirname with self.directory should work.

If you try out e.g. loading an image in an image texture node in a shader you should see that Blender always opens the first of the selected files, regardless of what self.filepath is.  If this isn't the case on your system then we probably need to account for OS differences.

Reply:
In the 2. commit (Better solution for open image with relative path failing) I changed it back and removed that line.